### PR TITLE
[RISCV] Fix bugs about getting register list of Zcmp push/pop.

### PR DESCRIPTION
**Pull request by yetingk**
The pr does two things, one is to fix internal compiler error when we need to spill callee saves but none of them is GPR, another is to fix wrong register number for pushed registers are {ra, s0-s11}.

### CODE REVIEW

**Comment by jrtc27** [comment]
ICE is GCC terminology

File: llvm/lib/Target/RISCV/RISCVFrameLowering.cpp [hunk 1]
```diff
@@ -228,35 +228,36 @@ getRestoreLibCallName(const MachineFunction &MF,
 
 // Return encoded value for PUSH/POP instruction, representing
 // registers to store/load.
```
Comment (kito-cheng) [inline llvm/lib/Target/RISCV/RISCVFrameLowering.cpp:230]: Comment need to update to reflect the update.

File: llvm/test/CodeGen/RISCV/callee-saved-gprs.ll [hunk 2]
```diff
@@ -2078,53 +2078,53 @@ define void @foo() {
...
 ; RV64IZCMP-WITH-FP-NEXT:    #NO_APP
 ; RV64IZCMP-WITH-FP-NEXT:    ld ra, 24(sp) # 8-byte Folded Reload
 ; RV64IZCMP-WITH-FP-NEXT:    ld s0, 16(sp) # 8-byte Folded Reload
-; RV64IZCMP-WITH-FP-NEXT:    ld s4, 8(sp) # 8-byte Folded Reload
+; RV64IZCMP-WITH-FP-NEXT:    ld s11, 8(sp) # 8-byte Folded Reload
 ; RV64IZCMP-WITH-FP-NEXT:    addi sp, sp, 32
 ; RV64IZCMP-WITH-FP-NEXT:    ret
 entry:
-  tail call void asm sideeffect "li s4, 0", "~{s4}"()
+  tail call void asm sideeffect "li s11, 0", "~{s11}"()
```
Comment (kito-cheng) [inline llvm/test/CodeGen/RISCV/callee-saved-gprs.ll:2127]: Add one more function instead of updating this?

**Review by asb — APPROVED** [review]
I believe this is correct, would appreciate another quick review from someone who's been more active in the other zcmp patches though.

**Pull request merged**